### PR TITLE
Updated materials editor

### PR DESCRIPTION
**Pull request by kritika-singh3**
Issue: #7847 

Description:
 - added a switch binded to `auto_update` for scm fields - defaulted the value to true
 - added a switch binded to `ignore_for_scheduling` for dependency material - default value false

### CODE REVIEW

**Comment by kritika-singh3** [comment]
SCM Materials:

Screen Shot 2020-03-13 at 3 21 10 PM

Dependency Material:

Screen Shot 2020-03-13 at 3 21 21 PM

**Comment by kritika-singh3** [comment]
Error on `auto_update`:

Screen Shot 2020-03-13 at 3 23 13 PM

File: server/src/main/webapp/WEB-INF/rails/webpack/views/components/switch/index.tsx [hunk 1]
```diff
@@ -59,14 +60,15 @@ export class SwitchBtn extends MithrilViewComponent<Attrs> {
                    vnode.attrs.field(target.checked);
                  }}
                  class={styles.switchInput}
-                 data-test-id="switch-checkbox"/>
+                 data-test-id={dataTestId}/>
```
Comment (marques-work) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/components/switch/index.tsx]: I like this -- tests are more resilient and it communicates intent. 👍

Comment (kritika-singh3) [reply server/src/main/webapp/WEB-INF/rails/webpack/views/components/switch/index.tsx]: Thanks!

**Pull request merged**